service/controller/clusterapi/v27/resource/ipam/create.go

```diff
@@ -53,12 +59,12 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if subnet needs to be allocated for cluster")
 
-	if legacykey.StatusNetworkCIDR(cr) == "" {
+	if key.StatusClusterNetworkCIDR(cr) == "" {
 		var subnetCIDR net.IPNet
 		{
 			r.logger.LogCtx(ctx, "level", "debug", "message", "allocating cluster subnet CIDR")
 
-			randomAZs, err := r.selectRandomAZs(legacykey.SpecAvailabilityZones(cr))
+			randomAZs, err := r.selectRandomAZs(key.WorkerAvailabilityZones(cc.Status.TenantCluster.TCCP.MachineDeployment))
```

Comment:
I am not sure if this right though. We have AZ config in both specs of the Cluster and the MachineDeployment CR.

Reply:
No, this is not correct, but then again, we can't really make this correct. The thing here is that this is only executed when creating a cluster so at that point `controllercontext` cannot contain any AZs, but I think we can set `randomAZs` to be a single element slice with configured master AZ.

Reply:
I think you are out already. Let's talk tomorrow please.

Reply:
We chatted and sorted it out. Randomization will be done in the API since we defined the AZs already in the MachineDeployment's provider config spec.